src/com/google/cose/utils/CoseUtils.java

```diff
@@ -92,33 +97,35 @@ public static DataItem getValueFromMap(Map cborMap, int value) {
     return cborMap.get(key);
   }
 
-  /**
-   * Generates EC2 Private Key from d parameter.
-   *
-   * Only supports P256 curve currently.
-   * @param curve supported curve
-   * @param d BigInteger representation for the private key bytes.
-   * @return PrivateKey JCA implementation
-   * @throws CoseException if unsupported key curve is used.
-   */
-  public static PrivateKey getEc2PrivateKeyFromCoordinate(int curve, BigInteger d)
+  public static PrivateKey getEc2PrivateKeyFromEncodedKeyBytes(byte[] encodedPrivateKeyBytes)
```

Comment:
Can you please describe how this is encoded? I'm guessing the COSE RFC defines this.

Reply:
renamed the function to reflect pkcs8 encoding.